src/shared/sleep-config.c

```diff
@@ -210,11 +211,13 @@ int battery_is_low(void) {
         if (r < 0)
                 return log_debug_errno(r, "Failed to initialize battery enumerator: %m");
 
```

Comment:
just check if there's any devices with `sd_device_enumerator_get_device_first` here

Reply:
I was thinking it might not be worth it as `FOREACH_DEVICE` calls it anyway, and the condition would shortcut things for us 🤔